aten/src/ATen/native/native_functions.yaml

```diff
@@ -462,8 +462,10 @@
   device_check: NoCheck   # TensorIterator
   structured: True
   structured_inherits: TensorIteratorBase
+  ufunc_inner_loop:
+    Generic: add (AllAndComplex, BFloat16, Half)
+    ScalarOnly: add (Bool)
   dispatch:
-    CPU, CUDA: add_out
```

Comment:
Didn't dive into the codegen yet. What happens if both CPU and ufunc_inner_loop are specified?

Reply:
It doesn't seem to error, and the CPU dispatch key gets ignored. Seems worth adding more error checking for this case.